src/main/java/com/coreos/jetcd/Client.java

```diff
@@ -1,21 +1,24 @@
 package com.coreos.jetcd;
 
+import com.coreos.jetcd.exception.AuthFailedException;
+import com.coreos.jetcd.exception.ConnectException;
+
 /**
  * Etcd Client.
  */
 public interface Client {
 
-  Auth getAuthClient();
+  Auth getAuthClient() throws AuthFailedException, ConnectException;
```

Comment:
or just throw `EtcdException` instead of `AuthFailedException `  and `ConnectException`.